staging/src/k8s.io/apiserver/pkg/storage/etcd3/watcher.go

```diff
@@ -561,25 +566,23 @@ func (wc *watchChan) acceptAll() bool {
 }
 
 // transform transforms an event into a result for user if not filtered.
-func (wc *watchChan) transform(e *event) (res *watch.Event) {
+func (wc *watchChan) transform(e *event) (res *watch.Event, err error) {
 	curObj, oldObj, err := wc.prepareObjs(e)
 	if err != nil {
 		klog.Errorf("failed to prepare current and previous objects: %v", err)
-		wc.sendError(err)
-		return nil
+		return nil, err
```

Comment:
should we wrap the error as `fmt.Errorf("failed to prepare current and previous objects: %w", err)`?

Reply:
done